DEVELOPING.md

```diff
@@ -0,0 +1,67 @@
+# For Developers
+
+We strongly encourage you to review the project's scope described in the `README.md` file before working on new features. For large changes, consider writing a design document using [this template](https://docs.google.com/document/d/1U6GlAaPF3tl1wvBNXebW1mD3oHZxV5h9TH-bR0p8Uz8/edit?usp=sharing).
+
+## Testing changing downstream
+
+By default, downstream projects use both an `http_archive` of `build_bazel_rules_typescript` and the released `@bazel/typescript` and `@bazel/karma` npm packages. `postinstall` steps in these npm packages check that the version of the `build_bazel_rules_typescript` is compatible with the version of the npm package(s).
+
+For example, if a downstream `WORKSPACE` contain:
+
+```python
+http_archive(
+    name = "build_bazel_rules_typescript",
+    url = "https://github.com/bazelbuild/rules_typescript/archive/0.21.0.zip",
+    strip_prefix = "rules_typescript-0.21.0",
+)
+```
+
+that that project's `package.json` would contain the matching:
+
+```json
+"@bazel/typescript": "0.21.0",
+"@bazel/karma": "0.21.0",
+```
+
+When authoring changes and testing downstream, depending on the `@bazel/typescript` and `@bazel/karma` npm packages makes the workflow confusing and difficult.
+To make authoring and testing changes downstream easier, it is recommended that you override the default `compiler` attribute of `ts_library` if making changes
+to `ts_library` and the default `karma` attribute of `ts_web_test_suite`/`ts_web_test` if making changes to those rules.
+
+For example, in `/internal/build_defs.bzl`, change
+
+```python
+"compiler": attr.label(
+    default = Label(_DEFAULT_COMPILER),
+```
+
+to
+
+```python
+"compiler": attr.label(
+    default = Label("@build_bazel_rules_typescript//internal:tsc_wrapped_bin"),
+```
+
+The correct defaults to use so that you are not depending on the npm package downstream are in `/internal/defaults.bzl`. Note, your downstream
+workspace will also need the correct `@npm` dependencies available to build these targets (see `internal/e2e/package_typescript_3.1_no_npm/package.json`).
+In the case of the `angular` workspace, some `@npm` dependencies in this repository will also need to be changed to `@ngdeps` since `angular` does not have
```

Comment:
At first it's somewhat surprising that this is needed, but it makes sense. 

The npm package installed downstream was using the `typescript` npm peerDependency. So we need to tell the rule to do the same.

I wonder if there's a more ergonomic way of doing this, like saving `"npm"` in a variable then using something like `"@%s//typescript" % npmLabel`,". I'm not sure of who would be doing this besides us though.